internal/datastore/postgres/options.go

```diff
@@ -107,52 +106,112 @@ func SplitAtUsersetCount(splitAtUsersetCount uint16) Option {
+	return func(po *postgresOptions) { po.readPoolOpts.ConnMaxIdleTime = &idle }
 }
 
-// ConnMaxLifetime is the duration since creation after which a connection will
-// be automatically closed.
+// WriteConnMaxIdleTime is the duration after which an idle write connection
+// will be automatically closed by the health check.
 //
 // This value defaults to having no maximum.
-func ConnMaxLifetime(lifetime time.Duration) Option {
-	return func(po *postgresOptions) {
-		po.connMaxLifetime = &lifetime
-	}
+func WriteConnMaxIdleTime(idle time.Duration) Option {
+	return func(po *postgresOptions) { po.writePoolOpts.ConnMaxIdleTime = &idle }
 }
 
-// HealthCheckPeriod is the interval by which idle Postgres client connections
-// are health checked in order to keep them alive in a connection pool.
-func HealthCheckPeriod(period time.Duration) Option {
-	return func(po *postgresOptions) {
-		po.healthCheckPeriod = &period
-	}
+// ReadConnMaxLifetime is the duration since creation after which a read
+// connection will be automatically closed.
+//
+// This value defaults to having no maximum.
+func ReadConnMaxLifetime(lifetime time.Duration) Option {
+	return func(po *postgresOptions) { po.readPoolOpts.ConnMaxLifetime = &lifetime }
 }
 
-// MaxOpenConns is the maximum size of the connection pool.
+// WriteConnMaxLifetime is the duration since creation after which a write
+// connection will be automatically closed.
 //
 // This value defaults to having no maximum.
-func MaxOpenConns(conns int) Option {
-	return func(po *postgresOptions) {
-		po.maxOpenConns = &conns
-	}
+func WriteConnMaxLifetime(lifetime time.Duration) Option {
+	return func(po *postgresOptions) { po.writePoolOpts.ConnMaxLifetime = &lifetime }
 }
 
-// MinOpenConns is the minimum size of the connection pool.
+// MinOpenReadConns is the minimum size of the connection pool used for reads.
+//
 // The health check will increase the number of connections to this amount if
 // it had dropped below.
 //
 // This value defaults to zero.
-func MinOpenConns(conns int) Option {
-	return func(po *postgresOptions) {
-		po.minOpenConns = &conns
-	}
+func MinOpenReadConns(conns int) Option {
+	return func(po *postgresOptions) { po.readPoolOpts.MinOpenConns = &conns }
+}
+
+// MinOpenWriteConns is the minimum size of the connection pool used for writes.
+//
+// The health check will increase the number of connections to this amount if
+// it had dropped below.
+//
+// This value defaults to zero.
+func MinOpenWriteConns(conns int) Option {
+	return func(po *postgresOptions) { po.writePoolOpts.MinOpenConns = &conns }
+}
+
+// MaxOpenReadConns is the maximum size of the connection pool used for reads.
+//
+// This value defaults to having no maximum.
+func MaxOpenReadConns(conns int) Option {
+	return func(po *postgresOptions) { po.readPoolOpts.MaxOpenConns = &conns }
+}
+
+// MaxOpenWriteConns is the maximum size of the connection pool used for writes.
+//
+// This value defaults to having no maximum.
```

Comment:
Is this correct? I see defaults set in the CLI to:

```go
func DefaultConnPoolConfig() *ConnPoolConfig {
	return &ConnPoolConfig{
		MaxLifetime:         30 * time.Minute,
		MaxIdleTime:         30 * time.Minute,
		MaxOpenConns:        20,
		MinOpenConns:        10,
		HealthCheckInterval: 30 * time.Second,
	}
}
```

Reply:
^ yes, it always defaults to 20, noticed this when I was testing it out

Reply:
What do we want these defaults to be? I think write pools should always default to smaller than the read pools.
It could also make sense that these be different based on the datastore.

Reply:
Sorry -- I hijacked this with a different question

If you don't use this option to configure the datastore, it defaults to having no maximum.

Reply:
The write pool now defaults to half the read pool.